### PR TITLE
8. User Can Complete Order - Complete Functionality

### PR DESCRIPTION
**Pull request by jnelsontn**
## Status
Ready

## Description
This pull request completes the ticket entitled "User Can Complete Order." It adds functionality to create an order, add a product to an order and complete an order.

## Related Tickets 

[https://github.com/illegal-llamas/djangazon/issues/2](https://github.com/illegal-llamas/djangazon/issues/2)

## Impacted Areas of the Application
```
website/forms.py
website/models.py
website/templates/cart.html
website/templates/checkout.html
website/templates/login.html
website/templates/single.html
website/urls.py
website/views.py
```

## Deploy Notes
To test complete functionality - You will need to sign up a customer, add payment types, add products to your cart and checkout to complete your order. Replicate multiple times with multiple customers to test full functionality.

For example: 
```
    git pull 
    git fetch --all
    git checkout jn-updates
    ./migrate_llamas.sh
   python manage.py runserver
```

### CODE REVIEW

**Comment by sarawithad** [comment]
Yaaaaay! Working for me. :shipit:

File: website/views.py [hunk 1]
```diff
@@ -241,51 +244,87 @@ def user_payment_types(request):
...
 
+@login_required(login_url='/login')
+def complete_order_add_payment(request, order_id):
+    """
+    purpose: Allows user to add a payment type to their order and therefore complete and place the order
+    author: Dara Thomas
+    args: order_id - no clue what to do with this
+    returns: a checkout page where the user sees their order total and can select a payment type for their order
+    """
+    if request.method == 'POST':
```
Comment (hfrankst) [inline website/views.py:303]: Add a check to see if a payment type has been added, if not then redirect the user to the add a payment type form.

**Review by hfrankst — CHANGES_REQUESTED** [review]

File: website/views.py (hunk 1 above)
Comment (jnelsontn) [reply website/views.py:303]: bitmoji

Comment (jnelsontn) [reply website/views.py:303]: This change has been implemented.

**Review by hfrankst — APPROVED** [review]

**Pull request merged**